src/app/downloads/downloads.component.spec.ts

```diff
@@ -6,75 +6,43 @@ import { StoreModule, Store } from '@ngrx/store';
 import { Subject } from 'rxjs/Subject';
 import { Angulartics2 } from 'angulartics2';
 
-import { MockHalService, MockHalDoc, AuthModule, FancyFormModule } from 'ngx-prx-styleguide';
+import { MockHalService, AuthModule, FancyFormModule } from 'ngx-prx-styleguide';
 import { CoreModule, CastleService } from '../core';
 import { SharedModule } from '../shared';
 import { DownloadsComponent } from './downloads.component';
-import { DownloadsChartComponent } from './downloads-chart.component';
-import { DownloadsTableComponent } from './downloads-table.component';
+import { DownloadsChartContainerComponent } from './downloads-chart-container.component';
+import { DownloadsChartPresentationComponent } from './downloads-chart-presentation.component';
+import { DownloadsTableContainerComponent } from './downloads-table-container.component';
+import { DownloadsTablePresentationComponent } from './downloads-table-presentation.component';
 import { downloadsRouting } from './downloads.routing';
 
-import { EpisodeModel, PodcastModel, RouterModel, ChartType, MetricsType,
-  CHARTTYPE_STACKED, METRICSTYPE_DOWNLOADS, INTERVAL_DAILY, getMetricsProperty } from '../ngrx';
+import { METRICSTYPE_DOWNLOADS, INTERVAL_DAILY, getMetricsProperty } from '../ngrx';
 import { reducers } from '../ngrx/reducers';
 import * as ACTIONS from '../ngrx/actions';
 
+import { routerState, episodes, podcast, ep0Downloads, ep1Downloads, podDownloads } from '../../testing/downloads.fixtures';
+
 describe('DownloadsComponent', () => {
   let comp: DownloadsComponent;
   let fix: ComponentFixture<DownloadsComponent>;
   let de: DebugElement;
   let el: HTMLElement;
   let castle;
   let store: Store<any>;
-
-  const podcast: PodcastModel = {
-    seriesId: 37800,
-    feederId: '70',
-    title: 'Pet Talks Daily'
-  };
-  podcast['doc'] = new MockHalDoc(podcast);
-  const episode: EpisodeModel = {
-    id: 1,
-    page: 1,
-    guid: 'abcde',
-    seriesId: 37800,
-    title: 'A Pet Talks Episode',
-    publishedAt: new Date()
-  };
-  const routerState: RouterModel = {
-    podcastSeriesId: 37800,
-      page: 1,
-      beginDate: new Date('2017-08-27T00:00:00Z'),
-      endDate: new Date('2017-09-07T00:00:00Z'),
-      interval: INTERVAL_DAILY,
-      chartType: <ChartType>CHARTTYPE_STACKED,
-      metricsType: <MetricsType>METRICSTYPE_DOWNLOADS
-  };
-  const downloads = [
-    ['2017-08-27T00:00:00Z', 52522],
-    ['2017-08-28T00:00:00Z', 162900],
-    ['2017-08-29T00:00:00Z', 46858],
-    ['2017-08-30T00:00:00Z', 52522],
-    ['2017-08-31T00:00:00Z', 162900],
-    ['2017-09-01T00:00:00Z', 46858],
-    ['2017-09-02T00:00:00Z', 52522],
-    ['2017-09-03T00:00:00Z', 162900],
-    ['2017-09-04T00:00:00Z', 46858],
-    ['2017-09-05T00:00:00Z', 52522],
-    ['2017-09-06T00:00:00Z', 162900],
-    ['2017-09-07T00:00:00Z', 46858]
-  ];
+  const metricsPropertyName = getMetricsProperty(routerState.interval, routerState.metricsType);
```

Comment:
Side note: I hate myself a little bit for letting that property name leak outside the reducer; should have used two separate params for the action.